Installation issue: Sz 2.1.8.3 GCC 9.3.0 Ubuntu 20.04

Using `spack@develop` with commit id `70c3b0ba0941` from 3-June-2020.

System:
* Ubuntu 20.04
* GCC 9.3.0
* x86_64 and ppc64le

```
$> spack install sz
...
2 errors found in build log:
     41    make  -f CMakeFiles/tags.dir/build.make CMakeFiles/tags.dir/build
     42    make[2]: Entering directory '/tmp/root/spack-stage/spack-stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcashretjt6/spack-src/spack-build'
     43    [  1%] files
     44    cd /tmp/root/spack-stage/spack-stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcashretjt6/spack-src && /usr/bin/ctags --exclude=/tmp/root/spack-stage/spack
           -stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcashretjt6/spack-src/spack-build -f /tmp/root/spack-stage/spack-stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcas
           hretjt6/spack-src/spack-build/tags --c++-kinds=+p --fields=+iaS -R
     45    /usr/bin/ctags: unrecognized option '--exclude=/tmp/root/spack-stage/spack-stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcashretjt6/spack-src/spack-build
           '
     46    	Try '/usr/bin/ctags --help' for a complete list of options.
  >> 47    make[2]: *** [CMakeFiles/tags.dir/build.make:81: CMakeFiles/tags] Error 1
     48    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcashretjt6/spack-src/spack-build'
  >> 49    make[1]: *** [CMakeFiles/Makefile2:173: CMakeFiles/tags.dir/all] Error 2
     50    make[1]: *** Waiting for unfinished jobs....
     51    Scanning dependencies of target zstd
     52    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcashretjt6/spack-src/spack-build'
     53    make  -f zstd/CMakeFiles/zstd.dir/build.make zstd/CMakeFiles/zstd.dir/build
     54    make[2]: Entering directory '/tmp/root/spack-stage/spack-stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcashretjt6/spack-src/spack-build'
     55    [  2%] Building C object zstd/CMakeFiles/zstd.dir/common/entropy_common.c.o

See build log for details:
  /tmp/root/spack-stage/spack-stage-sz-2.1.8.3-rl7uy4c4jokfzadoaah7crcashretjt6/spack-build-out.txt
```

Full build log: [sz.txt](https://github.com/spack/spack/files/4724710/sz.txt)
Configure args: [sz-configure.txt](https://github.com/spack/spack/files/4724714/sz-configure.txt)

@disheng222 @germasch @robertu94

--- a/var/spack/repos/builtin/packages/sz/ctags-only-if-requested.patch
+++ b/var/spack/repos/builtin/packages/sz/ctags-only-if-requested.patch
@@ -1,0 +1,42 @@
+From 3637a87f986cb64ddc1bde3551e29894ab18500b Mon Sep 17 00:00:00 2001
+From: Robert Underwood <rr.underwood94@gmail.com>
+Date: Wed, 3 Jun 2020 12:48:33 -0400
+Subject: [PATCH] Don't require ctags unless we ask for it
+
+Previously this caused build failures if the user had a bad version of
+ctags.  Don't build with it.
+---
+ CMakeLists.txt | 17 ++++++++++-------
+ 1 file changed, 10 insertions(+), 7 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 4fd4014..d764a44 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -36,13 +36,16 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+ 
+ #generate tags for the project if tags exist
+-find_program(TAGS ctags)
+-if(TAGS)
+-	add_custom_target(tags ALL
+-    COMMAND ${TAGS} --exclude=${CMAKE_BINARY_DIR} -f ${CMAKE_BINARY_DIR}/tags --c++-kinds=+p --fields=+iaS -R
+-		COMMENT Generating Tag files
+-		WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+-		)
++option(BUILD_CTAGS "enable ctags generation target" OFF)
++if(BUILD_CTAGS)
++  find_program(TAGS ctags)
++  if(TAGS)
++    add_custom_target(tags ALL
++      COMMAND ${TAGS} --exclude=${CMAKE_BINARY_DIR} -f ${CMAKE_BINARY_DIR}/tags --c++-kinds=+p --fields=+iaS -R
++      COMMENT Generating Tag files
++      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
++      )
++  endif()
+ endif()
+ 
+ option(BUILD_SHARED_LIBS "build shared libraries over static libraries" ON)
+-- 
+2.24.2 (Apple Git-127)
+

--- a/var/spack/repos/builtin/packages/sz/package.py
+++ b/var/spack/repos/builtin/packages/sz/package.py
@@ -54,6 +54,8 @@ class Sz(CMakePackage):
     depends_on('hdf5', when="+hdf5")
     depends_on('netcdf-c', when="+netcdf")
 
+    patch('ctags-only-if-requested.patch', when='@2.1.8.1:2.1.8.3')
+
     @property
     def build_directory(self):
         """autotools needs a different build directory to work"""

